setup.ts

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env -S deno run --unstable --allow-run --allow-read --allow-write
+
+import { exists } from "https://deno.land/std@0.103.0/fs/mod.ts";
+import * as shell from "./aggregator/programs/helpers/shell.ts";
+
+const components = [
+    { name: "aggregator", skipYarn: true },
+    { name: "clients" },
+    { name: "contracts" },
+    { name: "extension" },
+    { name: "signer" },
+];
+
+async function runYarn(name: string): Promise<void> {
+    const { success } = await Deno.run({ cmd: ["yarn"], cwd: `./${name}` }).status();
+    if (!success) {
+        throw new Error(`yarn install failed for ${name}`);
+    }
+}
+
+console.log("initializing bls-wallet repo");
+
+console.log("initializing git submodules...");
+await shell.run(..."git submodule update --init --recursive".split(" "));
+
+console.log(`setting up components (${components.join(", ")})...`);
+await Promise.all(components.map(async ({ name, skipYarn }) => {
+    if (!skipYarn) {
+        console.log(`yarn installing in ${name}...`);
+        await runYarn(name);
+    }
+
+    const envFilePath = `./${name}/.env`;
+    const envExampleFilePath = `${envFilePath}.example`;
+    console.log(`copying ${envExampleFilePath} to ${envFilePath}...`);
```

Comment:
Maybe move this inside the `if`, otherwise if you any existing `.env` files it would suggest they had been overwritten.